CareKitStore/CareKitStoreTests/OCKStore/TestStore+Sync.swift

```diff
@@ -266,6 +266,92 @@ class TestStoreSync: XCTestCase {
         XCTAssert(localOutcomes == remoteOutcomes)
         XCTAssert(localOutcomes.count == 1)
     }
+    
+    func testTombstoningOutcomePushedToRemote() throws {
+        let dummy2 = DummyEndpoint2()
+        let testStore = OCKStore(name: "test", type: .inMemory, remote: dummy2)
+        dummy2.automaticallySynchronizes = false
+        
+        let schedule = OCKSchedule.mealTimesEachDay(start: Date(), end: nil)
+        let task = try testStore.addTaskAndWait(OCKTask(id: "A", title: "A", carePlanUUID: nil, schedule: schedule))
+        let taskUUID = try task.getUUID()
+
+        let outcome = try testStore.addOutcomeAndWait(OCKOutcome(taskUUID: taskUUID, taskOccurrenceIndex: 0, values: [OCKOutcomeValue("test")]))
+        let outcomeUUID = try outcome.getUUID()
+        XCTAssertNoThrow(try testStore.syncAndWait()) //Sync original outcome
+        
+        try testStore.deleteOutcomeAndWait(outcome)
+        XCTAssertNoThrow(try testStore.syncAndWait()) //Sync tombstoned outcome
+        let latestRevisions = dummy2.revisionsPushedInLastSynch
+        XCTAssert(latestRevisions.count == 2)
+        
+        let tombstonedOutcomes = latestRevisions.compactMap{
+            entity -> OCKOutcome? in
+            switch entity{
+            case .outcome(let outcome):
+                return outcome
+            default:
+                return nil
+            }
+        }
+        XCTAssert(tombstonedOutcomes.count == 2)
```

Comment:
I'm guessing that one of these is the new delete record, and the other is an "update" to the original to tombstone it?

If so, we should see if it's somehow possible to only send the new delete record. I think it should be possible for the client to know what to do given just the new delete record.

Reply:
Yup, I read this one after posting my previous comment, but this is essentially what I was trying to say about line 286.

I think the problem with knowing what to do on the client side is the change in UUID, it comes with a deleted date, but no reference to the UUID to delete.

If it was like the versioned objects, the tombstone could be the current version, and then the client would know to tombstone the previous version

Reply:
If it tombstoned the same UUID, I agree, it's easy to handle this on the remote

Reply:
I see where you're coming from. From CareKit's perspective, our diff generation logic is simpler if we don't also send over the tombstone, but I think you would make the argument that it's better to get it right once inside of CareKit than to ask each integration developer to figure out which outcome they need to tombstone.